### PR TITLE
fix(crons): Properly update + return environment for monitor alerts

### PR DESCRIPTION
**Pull request by rjo100**
Fills in gaps in existing environment alert coverage on the crons backend

### CODE REVIEW

File: src/sentry/monitors/models.py [hunk 1]
```diff
@@ -341,6 +341,8 @@ def get_alert_rule_data(self):
                     )
             alert_rule_data["targets"] = targets
 
+            alert_rule_data["environment_id"] = alert_rule.environment_id
```
Comment (evanpurkhiser) [inline src/sentry/monitors/models.py:344]: we need the environment name, not ID. Let's keep this consistent with what we've serialized

Comment (rjo100) [reply src/sentry/monitors/models.py:344]: hmmm...let me see what existing rules look like but pretty sure it just returns the id

Comment (evanpurkhiser) [reply src/sentry/monitors/models.py:344]: I'm saying for our API, we POST / PUT with the name, so in our response we also want to return the name right?

Comment (rjo100) [reply src/sentry/monitors/models.py:344]: ah...it uses a different serializer...

Comment (evanpurkhiser) [reply src/sentry/monitors/models.py:344]: I think this can just become `environment__name`

Comment (rjo100) [reply src/sentry/monitors/models.py:344]: using `environment`

Comment (rjo100) [reply src/sentry/monitors/models.py:344]: only the `id` is stored not an fk

**Comment by rjo100** [comment]
I do not understand why the typing test is failing

**Review by evanpurkhiser — APPROVED** [review]

**Pull request merged**